sdk/analytics/src/main/java/com/klaviyo/analytics/state/StateSideEffects.kt

```diff
@@ -0,0 +1,85 @@
+package com.klaviyo.analytics.state
+
+import com.klaviyo.analytics.model.ImmutableProfile
+import com.klaviyo.analytics.model.Keyword
+import com.klaviyo.analytics.model.PROFILE_ATTRIBUTES
+import com.klaviyo.analytics.model.ProfileKey
+import com.klaviyo.analytics.networking.ApiClient
+import com.klaviyo.analytics.networking.requests.ApiRequest
+import com.klaviyo.analytics.networking.requests.KlaviyoApiRequest
+import com.klaviyo.analytics.networking.requests.PushTokenApiRequest
+import com.klaviyo.core.Registry
+import com.klaviyo.core.config.Clock
+
+internal class StateSideEffects(
+    private val state: State = Registry.get<State>(),
+    private val apiClient: ApiClient = Registry.get<ApiClient>()
+) {
+    /**
+     * Debounce timer for enqueuing profile API calls
+     */
+    private var timer: Clock.Cancellable? = null
+
+    /**
+     * Pending batch of profile updates to be merged into one API call
+     */
+    private var pendingProfile: ImmutableProfile? = null
+
+    init {
+        apiClient.onApiRequest(false, ::afterApiRequest)
+        state.onStateChange { key: Keyword? ->
+            when (key) {
+                ProfileKey.PUSH_STATE -> onPushStateChange()
+                ProfileKey.PUSH_TOKEN -> { /* Token is a no-op, push changes are captured by push state */ }
+                PROFILE_ATTRIBUTES -> if (state.get(true).attributes.propertyCount() > 0) {
+                    onUserStateChange()
+                }
+                else -> onUserStateChange()
+            }
+        }
+    }
+
+    private fun onPushStateChange() {
+        if (!state.pushState.isNullOrEmpty()) {
+            state.pushToken?.let { apiClient.enqueuePushToken(it, state.get()) }
+        }
+    }
+
+    private fun onUserStateChange() {
+        val profile = state.get(true)
```

Comment:
```suggestion
        val profile = state.get(withAttributes = true)
```
prefer using keywords so its clearer when reading what the args are for